packages/@aws-cdk/aws-apigateway/lib/integrations/stepfunctions.ts

```diff
@@ -0,0 +1,237 @@
+import { Method } from '../method';
+import { AwsIntegration } from './aws';
+
+/**
+ * Options when configuring Step Functions integration with Rest API
+ */
+export interface StepFunctionsIntegrationOptions extends IntegrationOptions {
+  /**
+    * Action for the Step Functions integration. The list of supported API actions can be found
+    * on https://docs.aws.amazon.com/step-functions/latest/apireference/API_Operations.html
+    * @default 'StartSyncExecution'
+    */
+  readonly action: string;
+
+  /**
+   * Check if cors is enabled
+   * @default false
+   */
+  readonly corsEnabled?: boolean;
+
+  /**
+   * You can add requestContext (similar to input requestContext from lambda input)
+   * to the input. The 'requestContext' parameter includes account ID, user identity, etc.
+   * that can be used by customers that want to know the identity of authorized users on
+   * the state machine side. You can individually select the keys you want by setting them to true.
+   * The following code defines a REST API like above but also adds 'requestContext' to the input
+   * of the State Machine:
+   *
+   * @example
+   *
+   *    const stateMachine = new stepFunctions.StateMachine(this, 'StateMachine', ...);
+   *    new apigateway.StepFunctionsRestApi(this, 'StepFunctionsRestApi', {
+   *      stateMachine: stateMachine,
+   *      requestContext: {
+   *         accountId: true,
+   *         apiId: true,
+   *         apiKey: true,
+   *         authorizerPrincipalId: true,
+   *         caller: true,
+   *         cognitoAuthenticationProvider: true,
+   *         cognitoAuthenticationType: true,
+   *         cognitoIdentityId: true,
+   *         cognitoIdentityPoolId: true,
+   *         httpMethod: true,
+   *         stage: true,
+   *         sourceIp: true,
+   *         user: true,
+   *         userAgent: true,
+   *         userArn: true,
+   *         requestId: true,
+   *         resourceId: true,
+   *         resourcePath: true,
+   *       },
+   *    });
+   *
+   * @default - all parameters within request context will be set as false
+   */
+  readonly requestContext?: RequestContext;
+
+
+}
+/**
+ * Integrates a Synchronous Express State Machine from AWS Step Functions to an API Gateway method.
+ *
+ * @example
+ *
+ *    const stateMachine = new sfn.StateMachine(this, 'MyStateMachine', ...);
+ *    api.addMethod('GET', new StepFunctionsIntegration(stateMachine));
+ */
+export class StepFunctionsIntegration extends AwsIntegration {
+  private readonly stateMachine: sfn.IStateMachine;
+  private readonly action: string;
+  constructor(stateMachine: sfn.IStateMachine, options: StepFunctionsIntegrationOptions = { action: 'StartSyncExecution' }) {
+    let requestContextRequested: boolean = (options.requestContext) ? true: false;
+
+    const integResponse = integrationResponse();
+    const requestTemplate = requestTemplates(stateMachine, requestContextRequested, options.requestContext);
```

Comment:
Change to

```
    const requestTemplate = requestTemplates(stateMachine, options.requestContext);
```

The boolean check can be done inside the `requestTemplate()` method.